opt/scopeserver/dataserver/utils/loom.py

```diff
@@ -397,19 +416,16 @@ def add_user_clustering(self, request) -> Tuple[bool, str]:
 
         new_clusterings = Loom.dfToNamedMatrix(clusterings)
 
-        self.loom_connection = self.lfh.change_loom_mode(
-            self.file_path, mode="r+", partial_md5_hash=self.partial_md5_hash
-        )
+        self.loom_connection = self.lfh.change_loom_mode(self.file_path, mode="r+")
         loom = self.loom_connection
         loom.ca.Clusterings = new_clusterings
         loom.attrs["MetaData"] = json.dumps(metaJson)
-        self.loom_connection = self.lfh.change_loom_mode(
-            self.file_path, mode="r", partial_md5_hash=self.partial_md5_hash
-        )
-        loom = self.loom_connection
+        self.loom_connection = self.lfh.change_loom_mode(self.file_path, mode="r")
+
+        self.update_ss(update="clusterings")
 
         try:
-            new_clustering_data = self.get_clustering_by_id(new_clustering_meta["id"])
+            _ = self.get_clustering_by_id(new_clustering_meta["id"])
```

Comment:
The function attempts to get the newly added id. We've had issues in the past where despite everything looking OK, the file didn't actually accept changes, so this is a sanity check to make sure the new clustering is actually present and readable.

Reply:
You'll have to excuse my ignorance but I'm not seeing a sanity check here. I just see a NO-OP: get a clustering but don't use it.

Reply:
I see what you're saying. The sanity check was relying on the fact that the getter will fail if the ID doesn't exist in the file. However, we can make this a more explicit check rather than relying on the failure.

Reply:
Yes please, especially since there's nothing in the definition of `get_clustering_by_id` to indicate that it might fail or how it might fail. I presume now that it's with an `IndexError`?

Reply:
Indeed, I will rewrite it to check the presence of the index specifically 👍